runtime/near-evm-runner/src/lib.rs

```diff
@@ -299,6 +358,58 @@ impl<'a> EvmContext<'a> {
             .ok_or(VMLogicError::EvmError(EvmError::IntegerOverflow))?;
         self.gas_counter.pay_action_accumulated(burn_gas, use_gas, ActionCosts::new_receipt)
     }
+
+    fn pay_gas_from_evm_gas(&mut self, op: EvmOpForGas) -> Result<()> {
+        let fee_cfg = &self.fees_config.evm_config;
+        let evm_gas = self.evm_gas_counter.used_gas.as_u64();
```

Comment:
Can this overflow?

Reply:
It cannot because it's less the prepaid_gas passed to evm, which is a u64. If ever used_gas become >= prepaid gas during evm, internally evm will revert txn because out of gas